action_map.js

```diff
@@ -136,4 +136,16 @@ class ActionMap {
     ret.height = viewbox.height;
     return ret;
   }
+  
+  chooseMap(){
```

Comment:
If you name this function `createMap` instead of `chooseMap`, you can move the other three map-making calls here as well, and just do `let map = this.createMap()` in the constructor. This will make the responsibilities clearer, and probably work well in the future as well when the map dimensions will vary based on the map.